avaframe/ana5Hybrid/hybridTools.py

```diff
@@ -0,0 +1,416 @@
+    and a few previous ones but discardings the one that are too close ( 2* csz < distFromLast <= 30 * csz).
+    Extend in this diretion for a distance 0.2 * length of the path.
+
+    Parameters
+    -----------
+    dem: dict
+        dem dict
+    profile: dict
+        profile to extend
+
+    Returns
+    --------
+    profile: dict
+        extended profile
+    """
+    header = dem['header']
+    xllc = header['xllcenter']
+    yllc = header['yllcenter']
+    csz = header['cellsize']
+    zRaster = dem['rasterData']
+    # get last point
+    xLast = profile['x'][-1]
+    yLast = profile['y'][-1]
+    zLast = profile['z'][-1]
+    sLast = profile['s'][-1]
+    # compute distance from last point:
+    r = DFAtls.norm(profile['x']-xLast, profile['y']-yLast, profile['z']-zLast)
+    # find the previous points
+    pointsOfInterestLast = np.where((r < 20 * csz) & (r > 2* csz))[0]
+    xInterest = profile['x'][pointsOfInterestLast]
+    yInterest = profile['y'][pointsOfInterestLast]
+    zInterest = profile['z'][pointsOfInterestLast]
+    # find the direction in which we need to extend the path
+    vDirX = xLast - xInterest
+    vDirY = yLast - yInterest
+    vDirZ = zLast - zInterest
+    vDirX, vDirY, vDirZ = DFAtls.normalize(np.array([vDirX]), np.array([vDirY]), np.array([vDirZ]))
+    vDirX = np.sum(vDirX)
+    vDirY = np.sum(vDirY)
+    vDirZ = np.sum(vDirZ)
+    vDirX, vDirY, vDirZ = DFAtls.normalize(np.array([vDirX]), np.array([vDirY]), np.array([vDirZ]))
+    # extend in this direction
+    gamma = 0.2 * sLast / np.sqrt(vDirX**2 + vDirY**2)
+    xExtBottom = np.array([xLast + gamma * vDirX])
+    yExtBottom = np.array([yLast + gamma * vDirY])
+    # project on cszDEM
+    zExtBottom, _ = geoTrans.projectOnGrid(xExtBottom, yExtBottom, zRaster, csz=csz, xllc=xllc, yllc=yllc, interp='bilinear')
+    # extend profile
+    profile['x'] = np.append(profile['x'], xExtBottom)
+    profile['y'] = np.append(profile['y'], yExtBottom)
+    profile['z'] = np.append(profile['z'], zExtBottom)
+
+    if debug:
+        fig, ax = plt.subplots(figsize=(pU.figW, pU.figH))
+        ax.set_title('Extend path towards the bottom')
+        ax.plot(profile['x'][:-1], profile['y'][:-1], '.k', label='mass averaged path')
+        ax.plot(xInterest, yInterest, '.m', markersize=10, label='points considered to find drection')
+        ax.plot(xInterest[-1], yInterest[-1], '.b', markersize=10, label='bottom point of the mass averaged path')
+        ax.plot(profile['x'][0], profile['y'][0], '.g', label='point in the extention direction at distance \n 0.2 x path length from the bottom point')
+        ax.plot(profile['x'][-2:], profile['y'][-2:], 'k--', label='extended path')
+        plt.legend()
+        plt.show()
+    return profile
+
+
+def plotHybridRes(avalancheDir, resAB, resABNew, name, pathDict, simID, rasterTransfo, resAnalysis, dem, demOri,
+                  particlesList, fieldsList, avaProfileMass, avaProfileMassNew):
```

Comment:
please make this function a bit more readable by adding comments about what is being plotted